pallets/relayer/src/lib.rs

```diff
@@ -0,0 +1,237 @@
+// This file is part of the Polymesh distribution (https://github.com/PolymathNetwork/Polymesh).
+// Copyright (c) 2020 Polymath
+
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, version 3.
+
+// This program is distributed in the hope that it will be useful, but
+// WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+// General Public License for more details.
+
+// You should have received a copy of the GNU General Public License
+// along with this program. If not, see <http://www.gnu.org/licenses/>.
+
+//! # Simple Relayer Module
+//!
+//! TODO: Add description.
+//! TODO: Add PolyLimit
+//! TODO: Add support for `AuthorizationData::AddRelayerPayingKey` to `CddAndFeeDetails` in `pallets/runtime/*/src/fee_details.rs`
+
+#![cfg_attr(not(feature = "std"), no_std)]
+
+#[cfg(feature = "runtime-benchmarks")]
+pub mod benchmarking;
+
+use frame_support::{decl_error, decl_module, decl_storage, dispatch::DispatchResult, ensure};
+use frame_system::ensure_signed;
+use pallet_identity::{self as identity, PermissionedCallOriginData};
+pub use polymesh_common_utilities::traits::relayer::{
+    Event, RawEvent, RelayerSubTrait, Trait, WeightInfo,
+};
+use polymesh_primitives::{AuthorizationData, IdentityId, Signatory};
+
+type Identity<T> = identity::Module<T>;
+
+decl_storage! {
+    trait Store for Module<T: Trait> as Relayer {
+        /// map `user_key` to `paying_key`
+        pub PayingKeys get(fn paying_keys):
+            map hasher(blake2_128_concat) T::AccountId => T::AccountId;
+    }
+}
+
+decl_module! {
+    pub struct Module<T: Trait> for enum Call where origin: T::Origin {
+        type Error = Error<T>;
+
+        fn deposit_event() = default;
+
+        /// Set paying key for `user_key`
+        /// TODO: Add docs.
+        #[weight = <T as Trait>::WeightInfo::set_paying_key()]
+        pub fn set_paying_key(origin, user_key: T::AccountId) -> DispatchResult {
+            Self::base_set_paying_key(origin, user_key)
+        }
```

Comment:
I think we can nix this based on today's tech call discussions.

Reply:
If we only keep `add_authorization` and don't provide per-pallet extrinsics, then how can a secondary_key/external agent be allowed to create some authorizations, but not others?

Also this one could be renamed to `authorize_paying_key`, the current name doesn't fit with what it does.